lib/id3c/cli/command/geocode.py

```diff
@@ -411,6 +412,34 @@ def parse_first_smartystreets_result(result: list) -> dict:
     return response
 
 
+def extract_address(address: dict) -> dict:
+    """
+    Given an *address*, converts it to text and returns a result from
+    the SmartyStreets US Extract API containing information about an address
+    connected to the text.
+
+    Note that this API is not consistent with the US Street API, and the lookup
+    and responses must be handled differently.
+    """
+    LOG.debug("Previous lookup failed. Looking up address as free text.")
+
+    client = smartystreets_client_builder().build_us_extract_api_client()
+    address_text = ', '.join([str(val) for val in list(address.values()) if val])
```

Comment:
This implicitly relies on the insertion order of keys in the address dictionary. That's ok, I think, since the CPython implementation did this since 3.6 (which we're using) and the Python language spec guarantees it since 3.7.

However, the insertion order comes from `address_column_map`, which puts `secondary` as the last key. We probably don't use this key, but if we did, I believe for address extraction to work it would have to come after `street`:

https://github.com/seattleflu/id3c/blob/36de719553321eb9be89a4745d3ed31eab7e9525/lib/id3c/cli/command/geocode.py#L202-L208

Reply:
Ah, yeah. I made sure they were in the right order within redcap-det etls but forgot to check within the geocode command itself 🤦‍♀